src/core/main.c

```diff
@@ -2373,8 +2373,13 @@ int main(int argc, char *argv[]) {
         m->timestamps[MANAGER_TIMESTAMP_KERNEL] = kernel_timestamp;
         m->timestamps[MANAGER_TIMESTAMP_INITRD] = initrd_timestamp;
         m->timestamps[MANAGER_TIMESTAMP_USERSPACE] = userspace_timestamp;
-        m->timestamps[MANAGER_TIMESTAMP_SECURITY_START] = security_start_timestamp;
-        m->timestamps[MANAGER_TIMESTAMP_SECURITY_FINISH] = security_finish_timestamp;
+        if (in_initrd()) {
+                m->timestamps[MANAGER_TIMESTAMP_INITRD_SECURITY_START] = security_start_timestamp;
+                m->timestamps[MANAGER_TIMESTAMP_INITRD_SECURITY_FINISH] = security_finish_timestamp;
+        } else {
+                m->timestamps[MANAGER_TIMESTAMP_SECURITY_START] = security_start_timestamp;
+                m->timestamps[MANAGER_TIMESTAMP_SECURITY_FINISH] = security_finish_timestamp;
```

Comment:
i wonder if we can find a nicer syntax for this. maybe add a new function like this:

```c
static ManagerTimestamp manager_timestamp_initrd_mangle(ManagerTimestamp s) {
    if (in_initrd() &&
            s >= MANAGER_TIMESTAMP_SECURITY_START &&
            s <= MANAGER_TIMESTAMP_UNITS_LOAD_FINISH)
                return s - MANAGER_TIMESTAMP_SECURITY_START + MANAGER_TIMESTAMP_INITRD_SECURITY_START;
        return s;
}
```

This basically maps the non-initrd version of these timestamp ids to the initrd version. We could then simply wrap the array index in the code above in a call to this function, and things would be a bit prettier I think